core/chains/evm/gas/models.go

```diff
@@ -362,6 +362,9 @@ func (e *evmFeeEstimator) estimateFeeLimit(ctx context.Context, feeLimit uint64,
 		To:   toAddress,
 		Data: calldata,
 	}
+	if fromAddress != nil {
```

Comment:
Shouldn't be treated the same way we treat `toAddress` ?

Reply:
We can't because of `callMsg`'s typing
```
type CallMsg struct {
	From  common.Address
	To    *common.Address
}
```